lib/router/src/lib.rs

```diff
@@ -72,30 +82,63 @@ where
 {
     recognize: Rec,
     make: Mk,
-    cache: Mutex<Cache<Rec::Target, LoadShed<Mk::Value>>>,
+    cache: Lock<Cache<Rec::Target, LoadShed<Mk::Value>>>,
 }
 
-enum State<Req, Svc>
+struct RouteRequest<Req, Rec, Mk>
```

Comment:
nit: I'd be inclined to embed this into the enum directly to reduce boilerplate... e.g.

```rust
enum State<Req, Rec, Mk>
where ...
{
    Inserting {
        request: Option<Req>,
        target: Option<Rec::Target>,
        make: Option<Mk>,
        cache: Lock<Cache<Rec::Target, LoadShed<Mk::Value>>>
    }
    ...
}
```